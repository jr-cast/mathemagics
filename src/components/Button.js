const Button = ({ className, text, onClick }) => {
  return <button className={className} type="button" text={text} onClick={onClick}>{text}</button>
}

export default Button;