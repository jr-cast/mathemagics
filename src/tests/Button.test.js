import React from "react";
import renderer from 'react-test-renderer';
import Button from "../components/Button";

const compute = () => {
  console.log('mock func');
}

test('renders Button component correctly', () => {
  const tree = renderer.create(<Button className="test1" text='text-test1' onClick={compute} />).toJSON();
  expect(tree).toMatchSnapshot();
});
