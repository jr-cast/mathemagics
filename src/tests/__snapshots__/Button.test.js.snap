// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component correctly 1`] = `
<button
  className="test1"
  onClick={[Function]}
  text="text-test1"
  type="button"
>
  text-test1
</button>
`;
